components/org.wso2.carbon.identity.local.auth.emailotp/src/main/java/org/wso2/carbon/identity/local/auth/emailotp/EmailOTPAuthenticator.java

```diff
@@ -1910,4 +1985,50 @@ private String getAssociatedUserLocale(AuthenticatedUser authenticatedUser, Auth
         }
         return null;
     }
+
+    protected int getMaximumResendAttempts(String tenantDomain, AuthenticationContext context)
+            throws AuthenticationFailedException {
+
+        try {
+            String allowedResendCount = AuthenticatorUtils.getEmailAuthenticatorConfig(
+                    AuthenticatorConstants.ConnectorConfig.EMAIL_OTP_RESEND_ATTEMPTS_COUNT, tenantDomain);
+
+            if (NumberUtils.isNumber(allowedResendCount)) {
```

Comment:
I noticed that Apache Commons Lang plans to deprecate `isNumber` in a future release. Therefore, can we catch the `NumberFormatException` from `parseInt` here and return a default value instead?